src/Microsoft.IdentityModel.Protocols/Configuration/HttpDocumentRetriever.cs

```diff
@@ -53,17 +54,12 @@ public HttpDocumentRetriever(HttpClient httpClient)
                 throw LogHelper.LogArgumentNullException("httpClient");
 
             _httpClient = httpClient;
-            _requireHttps = true;
```

Comment:
For complete correctness, we may want to think about distinguishing between our http client and this one passed in. The owner of httpClient could call dispose after calling HttpDocumentRetriever(HttpClient httpClient). I agree is it very unlikely and indicates a logic programming error. However, our logic could be: (local)httpclient = _httpClient ?? _defaultHttpClient.